src/xstate/AudioPlayerMachineContext.tsx

```diff
@@ -0,0 +1,74 @@
+/* eslint-disable import/prefer-default-export */
+import { createContext, useEffect } from 'react';
+
+import { useInterpret } from '@xstate/react';
+import { InterpreterFrom } from 'xstate';
+
+import { audioPlayerMachine } from './actors/audioPlayer/audioPlayerMachine';
+import {
+  getXstateStateFromLocalStorage,
+  persistXstateToLocalStorage,
+} from './actors/audioPlayer/audioPlayerPersistHelper';
+
+import { getAudioPlayerStateInitialState } from 'src/redux/defaultSettings/util';
+import { getUserPreferences } from 'src/utils/auth/api';
+import { isLoggedIn } from 'src/utils/auth/login';
+import PreferenceGroup from 'types/auth/PreferenceGroup';
+
+export const AudioPlayerMachineContext = createContext(
+  {} as InterpreterFrom<typeof audioPlayerMachine>,
+);
+
+export const AudioPlayerMachineProvider = ({ children, locale }) => {
+  const initialXstateContext = getXstateStateFromLocalStorage();
+  const defaultAudioStateByLocale = getAudioPlayerStateInitialState(locale);
+  const defaultLocaleContext = {
+    reciterId: defaultAudioStateByLocale.reciter.id,
+  };
+
+  const isClient = !!(
+    typeof window !== 'undefined' &&
+    window.document &&
+    window.document.createElement
+  );
+
+  const loggedIn = isLoggedIn();
+
+  const audioPlayerService = useInterpret(
+    audioPlayerMachine,
+    {
+      context: {
+        ...audioPlayerMachine.initialState.context,
+        ...defaultLocaleContext,
+        ...initialXstateContext,
+      },
+    },
+    (state) => {
+      const { playbackRate, reciterId } = state.context;
+      if (state.event.type !== 'UPDATE_TIMING') console.log(state.value, state.event);
+      persistXstateToLocalStorage({ playbackRate, reciterId });
```

Comment:
Shouldn't this be called only when events that change the reciter or playbackRate are triggered?